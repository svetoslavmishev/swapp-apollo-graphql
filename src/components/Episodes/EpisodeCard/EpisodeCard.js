import React, { useContext } from "react";
import PropTypes from "prop-types";
import { useHistory } from "react-router-dom";
import {
  Card,
  CardActionArea,
  CardMedia,
  CardContent,
  Typography
} from "@material-ui/core";

import { ThemeContext } from "../../../themeContext";
import styles from "./EpisodesCardStyles";

const EpisodeCard = ({ episode: { id, image, title, openingCrawl } }) => {
  const { currentTheme } = useContext(ThemeContext);
  const classes = styles({ currentTheme });
  const history = useHistory();

  return (
    <Card
      className={classes.card}
      onClick={() => history.push(`episodes/${id}`)}
    >
      <CardActionArea>
        <CardMedia className={classes.media} image={image} title={title} />
        <CardContent>
          <Typography classes={{ root: classes.title }} variant="h6">
            {title}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            {openingCrawl}
          </Typography>
        </CardContent>
      </CardActionArea>
    </Card>
  );
};

EpisodeCard.propTypes = {
  episode: PropTypes.shapeOf({
    id: PropTypes.string,
    image: PropTypes.string,
    title: PropTypes.string,
    openingCrawl: PropTypes.string
  })
};

export default EpisodeCard;
