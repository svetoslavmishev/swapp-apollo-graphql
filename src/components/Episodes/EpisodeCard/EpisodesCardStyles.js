import { makeStyles } from '@material-ui/styles';

const styles = makeStyles(theme => ({
  card: {
    maxWidth: 300,
    margin: 30,
    background: ({ currentTheme }) =>
      theme.palette[currentTheme].Cards.backgroundColor,
    border: ({ currentTheme }) =>
      `1px solid ${theme.palette[currentTheme].Cards.borderColor}`
  },
  title: {
    fontFamily: 'StarWars',
    color: ({ currentTheme }) =>
      theme.palette[currentTheme].PrimaryHeading.fontColor,
    backgroundColor: ({ currentTheme }) =>
      theme.palette[currentTheme].Cards.backgroundColor
  },
  media: {
    height: 0,
    paddingTop: '56.25%'
  }
}));

export default styles;
