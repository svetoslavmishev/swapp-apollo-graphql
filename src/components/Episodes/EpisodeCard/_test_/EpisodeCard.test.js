import React from 'react';
import { shallow } from 'enzyme';
import { ThemeContext, theme } from '../../../../themeContext';
import EpisodeCard from '../EpisodeCard';

describe('<EpisodeCard />', () => {
  it('renders <EpisodeCard /> without error', () => {
    const wrapper = shallow(
      <ThemeContext.Provider value={theme}>
        <EpisodeCard />
      </ThemeContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
