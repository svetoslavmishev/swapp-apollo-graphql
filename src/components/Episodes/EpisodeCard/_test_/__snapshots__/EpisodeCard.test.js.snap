// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EpisodeCard /> renders <EpisodeCard /> without error 1`] = `ShallowWrapper {}`;
